File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/ui/calendar/CalendarFragmentModel.java
package com.byagowi.persiancalendar.ui.calendar;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class CalendarFragmentModel extends ViewModel {
    public final MutableLiveData<MonthFragmentUpdateCommand> monthFragmentsHandler = new MutableLiveData<>();
    public final MutableLiveData<Long> selectedDayLiveData = new MutableLiveData<>();
    public boolean isTheFirstTime = true;

    void monthFragmentsUpdate(MonthFragmentUpdateCommand command) {
        monthFragmentsHandler.postValue(command);
    }

    public void selectDay(long jdn) {
        selectedDayLiveData.postValue(jdn);
    }

    static public class MonthFragmentUpdateCommand {
        final public int target;
        final public boolean isEventsModification;
        final public long currentlySelectedJdn;

        MonthFragmentUpdateCommand(int target, boolean isEventsModification, long currentSelectedJdn) {
            this.target = target;
            this.isEventsModification = isEventsModification;
            this.currentlySelectedJdn = currentSelectedJdn;
        }
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/ui/calendar/CalendarFragmentModel.kt
package com.byagowi.persiancalendar.ui.calendar

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class CalendarFragmentModel : ViewModel() {
    val monthFragmentsHandler = MutableLiveData<MonthFragmentUpdateCommand>()
    val selectedDayLiveData = MutableLiveData<Long>()
    var isTheFirstTime = true

    fun monthFragmentsUpdate(command: MonthFragmentUpdateCommand) {
        monthFragmentsHandler.postValue(command)
    }

    fun selectDay(jdn: Long) {
        selectedDayLiveData.postValue(jdn)
    }

    class MonthFragmentUpdateCommand internal constructor(val target: Int, val isEventsModification: Boolean, val currentlySelectedJdn: Long)
}
